### PR TITLE
Issue 1230 simplify

### LINKED ISSUE #986: Simplify basic operations
Automatically make the following simplifications:
`a+0=a`
`a-0=a`
`a*1=a`
`a/1=a`
`a**1=a`
`a**0=1`
Others?

### LINKED ISSUE #1230: Simplify all objects on creation
Instead of just simplifying if constant.
See also #986

### PR DESCRIPTION
**Pull request by tinosulzer**
# Description

Automatically simplify expressions that can be simplified, e.g. `1*x` becomes `x`.
This can be bypassed by creating the binary operator explicitly, e.g. `pybamm.Multiplication(1, x)`

Fixes #1230 
Fixes #986 

## Type of change

Please add a line in the relevant section of [CHANGELOG.md](https://github.com/pybamm-team/PyBaMM/blob/master/CHANGELOG.md) to document the change (include PR #) - note reverse order of PR #s. If necessary, also add to the list of breaking changes.

- [ ] New feature (non-breaking change which adds functionality)
- [x] Optimization (back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)

# Key checklist:

- [x] No style issues: `$ flake8`
- [x] All tests pass: `$ python run-tests.py --unit`
- [x] The documentation builds: `$ cd docs` and then `$ make clean; make html`

You can run all three at once, using `$ python run-tests.py --quick`.

## Further checks:

- [x] Code is commented, particularly in hard-to-understand areas
- [x] Tests added that prove fix is effective or that feature works

### CODE REVIEW

File: tests/unit/test_expression_tree/test_unary_operators.py [hunk 1]
```diff
@@ -457,6 +456,15 @@ def test_x_average(self):
             self.assertEqual(av_a.children[0].integration_variable[0].domain, x.domain)
             self.assertEqual(av_a.domain, [])
 
+        # whole electrode domain is different as the divisioni by 1 gets simplified out
```
Comment (brosaplanella) [inline tests/unit/test_expression_tree/test_unary_operators.py:459]: Typo in "divisioni"

**Review by brosaplanella — APPROVED** [review]
Looks good to me, just a typo to fix. Thanks Tino!

**Review by brosaplanella — APPROVED** [review]
Looks great, thanks Tino! Great job :)

File: pybamm/expression_tree/symbol.py [hunk 2]
```diff
@@ -696,14 +947,17 @@ def evaluates_to_number(self):
...
-        result = self.evaluate_ignoring_errors()
-
-        if isinstance(result, numbers.Number) or (
-            isinstance(result, np.ndarray) and np.prod(result.shape) == 1
-        ):
-            return True
-        else:
-            return False
+        return self.shape_for_testing == ()
+        # result = self.evaluate_ignoring_errors()
```
Comment (rtimms) [inline pybamm/expression_tree/symbol.py:951]: can this commented code be removed?

**Review by rtimms — APPROVED** [review]
thanks @tinosulzer !

**Pull request merged**